external_parser/event_processors/joined_event.h

```diff
@@ -159,21 +160,37 @@ struct ccb_joined_event : public typed_joined_event {
   }
 
   void set_cost(v_array<example *> &examples, float reward,
-                size_t index = 0) const override {
-    if (examples.size() <= index) {
+                size_t slot_index = 0) const override {
+    if (examples.size() <= slot_index) {
       VW::io::logger::log_warn(
-          "trying to set index [{}] when there are [{}] examples", index,
+          "trying to set index [{}] when there are [{}] examples", slot_index,
           examples.size());
       return;
     }
-    if (examples[index]->l.conditional_contextual_bandit.type !=
-        CCB::example_type::slot) {
-      VW::io::logger::log_warn(
-          "trying to set index [{}] on a CCB non-slot example", index);
-      return;
+
+    size_t index = 0;
+    for (auto &example : examples) {
+      if (example->l.conditional_contextual_bandit.type !=
+          CCB::example_type::slot) {
+        index++;
+        continue;
+      }
     }
-    examples[index]->l.conditional_contextual_bandit.outcome->cost =
+
+    size_t example_index = index + slot_index;
+    if (example_index >= examples.size()) {
+      VW::io::logger::log_error(
```

Comment:
Looks like we are just logging error but still continuing execution, and will probably have problems on line 186 during indexing.

Reply:
yup, nice catch.